File: backend/src/main/java/com/tomy/artifyme/artwork/ArtworkController.java
package com.tomy.artifyme.artwork;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;

@RestController
@RequiredArgsConstructor
public class ArtworkController {

    private final ArtworkService artworkService;

    @PostMapping("/api/v1/artwork")
    public ResponseEntity<ArtworkResponse> saveArtwork(@RequestBody ArtworkRequest request) {
        return artworkService.saveArtwork(request);
    }

    @GetMapping("/api/v1/artworks")
    public ResponseEntity<PageResponse> getArtworksWithPagination(
    @RequestParam(value = "pageNumber", required = false, defaultValue = "1") Integer pageNumber,
    @RequestParam(value = "pageSize", required = false, defaultValue = "10") Integer pageSize,
    @RequestParam(value = "useremail", required = true) String userEmail) {
        return artworkService.getArtworksWithPagination(pageNumber, pageSize, userEmail);
    }

    @GetMapping("/api/v1/artwork")
    public ResponseEntity<Object> getArtworkFromId(
        @RequestParam(value = "id", required = true) String id
    ){
        return artworkService.getArtworkById(id);
    }

    @DeleteMapping("/api/v1/artwork")
    public ResponseEntity<Object> deleteArtworkById(@RequestParam(value = "id", required = true) String id) {
        return artworkService.deleteArtwork(id);
    }

     @PatchMapping("/api/v1/artwork")
    public ResponseEntity<Object> updateArtwork(@RequestParam(value = "id", required = true) String id, @RequestBody ArtworkRequest request) {
        // Call the updateArtwork method of the artworkService and return its response
        return artworkService.updateArtwork(id, request);
    }
}

File: backend/Controllers/ArtworkController.cs
using Microsoft.AspNetCore.Mvc;
using ArtifyMe.Services;
using ArtifyMe.Models;
using ArtifyMe.Models.DTOs;
using System.Threading.Tasks;
using ArtifyMe.Services.Interfaces;

namespace ArtifyMe.Controllers
{
    [ApiController]
    [Route("api/v1/[controller]")]
    public class ArtworkController : ControllerBase
    {
        private readonly IArtworkService _artworkService;

        public ArtworkController(IArtworkService artworkService)
        {
            _artworkService = artworkService;
        }

        [HttpPost]
        public async Task<IActionResult> SaveArtwork([FromBody] ArtworkRequest request)
        {
            // Validate the request
            if (!ModelState.IsValid)
            {
                return BadRequest(ModelState);
            }

            return await _artworkService.SaveArtwork(request);
        }

        [HttpGet("artworks")]
        public async Task<IActionResult> GetArtworksWithPagination(
            [FromQuery] string userEmail,
            [FromQuery] int pageNumber = 1,
            [FromQuery] int pageSize = 10)
        {
         
            if (string.IsNullOrWhiteSpace(userEmail))
            {
                return BadRequest(new PageResponse
                {
                    Message = "User email must be provided.",
                    IsNext = false,
                    Content = new List<Artwork>()
                });
            }

    // Call the service layer to get paginated artworks
    var response = await _artworkService.GetArtworksWithPagination(pageNumber, pageSize, userEmail);

    // Return the response as-is, ensuring IsNext and Content are preserved
    return response;
}


        [HttpGet("artwork")]
        public async Task<IActionResult> GetArtworkById([FromQuery] string id)
        {
            return await _artworkService.GetArtworkById(id);
        }

        [HttpDelete("artwork")]
        public async Task<IActionResult> DeleteArtworkById([FromQuery] string id)
        {
            return await _artworkService.DeleteArtwork(id);
        }

        [HttpPatch("artwork")]
        public async Task<IActionResult> UpdateArtwork([FromQuery] string id, [FromBody] ArtworkRequest request)
        {
            // Validate the request
            if (!ModelState.IsValid)
            {
                return BadRequest(ModelState);
            }

            return await _artworkService.UpdateArtwork(id, request);
        }
    }
}